File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.py
class Solution(object):
    def maxProduct(self, n):
        """
        :type n: int
        :rtype: int
        """
        arr = []
        while n:
            arr.append(n % 10)
            n //= 10
        arr.sort()
        return arr[-1] * arr[-2]

File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.cs
public class Solution {
    public int MaxProduct(int num) {
        var n = 0;
        var m = 0;
        for (; num > 0; ) {
            int r = num % 10;
            if (r >= n) {
                m = n;
                n = r;
            }
            else if (r > m)
                m = r;
            num /= 10;
        }
        return n * m;
    }
}